### ISSUE TITLE
<single-row> gives false positive, when <all-rows> fails for a single row result set

### ISSUE THREAD

**Issue by lukzas** [Seddryck/NBi#547]
Consider the following predicate for a single row numeric value equal 5 to be more than 20

```xml
<test name="numeric">
    <system-under-test>
      <resultSet>
        <query>
          <![CDATA[
        SELECT 5
      ]]>
        </query>
      </resultSet>
    </system-under-test>
    <assert>
      <single-row>
        <predicate operand="#0" type="numeric">
          <more-than>20</more-than>
        </predicate>
      </single-row>
    </assert>
  </test>
```

Above test is passed (and it should not), but when used with `<all-rows>` tag instead it is evaluated properly and fails. Using alias or name for column does not help. When `more-than` value is changed, so that predicate holds, both tests are passed.

Tried NBi 1.21 RCs up to latest 1.22 beta v4

**Comment by Seddryck** [comment]
I confirm the bug

**Comment by Seddryck** [comment]
Great thanks for reporting this nasty bug.

The issue was that the `single-row` was not deserialized and the test resulted in a no test without constraint (so no potential failure). 

I fixed the deserialization and added a warning when running a test without constraint.

The fix will be included in release 1.21 but is now part of 1.22 branch.

Fix available at https://ci.appveyor.com/project/Seddryck/nbi/builds/28399795/artifacts or in `1.22-beta0009`
